### ISSUE TITLE
[IOS] FFMpeg command not working with PNG files

### ISSUE THREAD

**Issue by ankitsiliconithub** [tanersener/mobile-ffmpeg#26]
@tanersener 
Hi , I want to merge PNG with a Video but not working,  whereas working fine with JPG !

Please check code for reference : 

     let video:String = (Bundle.main.resourceURL?.appendingPathComponent("videoTest.mp4").path)!
            let videoOutput = String(format:VAR_DOCUMENT_DIRECTORY_SAVE + "/videoTestOutput.mp4")
            let image:String = (Bundle.main.resourceURL?.appendingPathComponent("maxresdefault.png").path)! // when i use jpg file it's working fine !!
        
            let result = MobileFFmpeg.execute("-y -loop 1 -framerate 24 -t 10 -i \(image) -i \(video) -filter_complex [0]scale=432:432,setsar=1[im];[1]scale=432:432,setsar=1[vid];[im][vid]concat=n=2:v=1:a=0 \(videoOutput)")
            print(result)
          
            if result == 0
            {
                print("SUCCESS")
            }

**Comment by tanersener** [comment]
> Hi , I want to merge PNG with a Video but not working, whereas working fine with JPG !

Could you please give us more detail about 'not working'? Is there a crash, command failure or success where output is not what you expect?

**Comment by ankitsiliconithub** [comment]
There is NO Crash , Xcode stuck at below mentioned line and RAM being used in GBs !

        let result = MobileFFmpeg.execute("-y -loop 1 -framerate 24 -t 10 -i \(image) -i \(video) -filter_complex [0]scale=432:432,setsar=1[im];[1]scale=432:432,setsar=1[vid];[im][vid]concat=n=2:v=1:a=0 \(videoOutput)")

and next line 

print(result)

doesn't even execute !!

**Comment by tanersener** [comment]
Do you have any console output for your process?

**Comment by ankitsiliconithub** [comment]
No not ouput nothing just hang Xcode and i have to shutdown my mac

**Comment by tanersener** [comment]
So Xcode console is empty, am I right?

**Comment by ankitsiliconithub** [comment]
i m not sure because i have to shudown my mac after run this cmd in simulator if you need code than i can provide you

**Comment by tanersener** [comment]
No thank you, you've already posted your command in your first post.

Let me make a few tests.

**Comment by tanersener** [comment]
Made some tests with your command. Your command syntax is correct and works if png file is valid. But if you png file is not correct then the following error message is printed continuously.

Is it possible to check if the same error message is printed in your application?

```
...
Error while decoding stream #0:0: Generic error in an external library
inflate returned error -3
Error while decoding stream #0:0: Generic error in an external library
inflate returned error -3
Error while decoding stream #0:0: Generic error in an external library
inflate returned error -3
Error while decoding stream #0:0: Generic error in an external library
...
```

**Comment by ankitsiliconithub** [comment]
Ok thank you for your effort :)

I tried again with all kinds of png files (with alpha & without alpha) but same following error message is printed continuously !

Please can you send me your png which is working fine at your side ...

...
Error while decoding stream #0:0: Generic error in an external library
inflate returned error -3
Error while decoding stream #0:0: Generic error in an external library
inflate returned error -3
Error while decoding stream #0:0: Generic error in an external library
inflate returned error -3
Error while decoding stream #0:0: Generic error in an external library
...

**Comment by tanersener** [comment]
Can you please check your `Build Settings` in `Xcode` and see if the following two settings are set to NO?

<img width="850" alt="png_settings" src="https://user-images.githubusercontent.com/10158439/45798948-794c9f80-bcb4-11e8-8881-8c61789b283c.png">

**Comment by ankitsiliconithub** [comment]
Wow ! it's working fine now , Thanks for your great support 👍 

I have a question , 

I'm using whole 'mobile-ffmpeg-universal' for just overlapping an image on Video but then ipa size becomes too large (around 500mb) !! , please can you suggest how to reduce overall app size with FFMpeg ?!

Best regards

**Comment by tanersener** [comment]
OK, cool :+1: 

I've updated universal binaries published in [v2.1.1 release](https://github.com/tanersener/mobile-ffmpeg/releases/tag/v2.1.1). There are 4 binaries there, static ones are smaller in size and `full` package is the smallest of them. 

```
mobile-ffmpeg-full-2.1.1-ios-static-universal
mobile-ffmpeg-full-gpl-2.1.1-ios-static-universal
mobile-ffmpeg-full-2.1.1-ios-shared-universal
mobile-ffmpeg-full-gpl-2.1.1-ios-shared-universal
```

To decrease the size further, a smaller package `min`, `https`, `audio`, `video`, etc. should be used. They are not available as universal binaries, I can generate them but I'll need time.

Important question here is which package do you need? What is the output video codec used in your application?

PS: Also remember that IPA includes all architectures: `armv7`, `armv7s`, `armv64`, also `i386` and `x86_64` if you haven't stripped them. So the actual application size is smaller, 500 / architecture count.

**Comment by ankitsiliconithub** [comment]
I use codec 'libx264' or 'mpeg4' to process Mp4 videos

with ffmpeg,  ipa size is +200mb more , so my question is , how can i reduce ipa size to a greater extent ! and which package should I choose or if I could simply discard some of dylibs which would be those ?! if any 

I'm using ffmpeg just for video processing (png image overlay on video just like watermark)

also you mentioned that you can generate package for that purpose , that would be great ! how much time (approx) should I wait .....

thank you

**Comment by tanersener** [comment]
> I use codec 'libx264' or 'mpeg4' to process Mp4 videos
>
>with ffmpeg, ipa size is +200mb more , so my question is , how can i reduce ipa size to a greater extent ! and which package should I choose or if I could simply discard some of dylibs which would be those ?! if any

You need `min-gpl` package. I don't recommend manually discarding .dylib or .a files; though you can discard some files and perform some tests to see if you have any errors. 

> also you mentioned that you can generate package for that purpose , that would be great ! how much time (approx) should I wait .....

I need a few hours to generate them all.

**Comment by tanersener** [comment]
@ankitsiliconithub 

Shared and static universal binaries for all packages are just published in the [v2.1.1 IOS Universal Binaries](https://github.com/tanersener/mobile-ffmpeg/releases/tag/v2.1.1.universal) release. I recommend `min-gpl` package for your application.
